modules/claims_api/app/controllers/claims_api/v1/application_controller.rb

```diff
@@ -173,6 +188,15 @@ def veteran_from_headers(with_gender: false)
         )
         # Fail fast if mpi_record can't be found
         unless vet.mpi_record?
+          # Intentionally NOT calling the claims_v1_logging method here
+          # to avoid the infinite loop that calling target_veteran there can create
+          # While we technically can just pass in icn: nil when doing that I felt it was best
+          # to avoid it entirely
+          ClaimsApi::Logger.log('unable_to_locate_mpi_record',
+                                message: 'unable_to_locate_mpi_record on request in v1 application controller.',
+                                api_version: 'V1',
+                                level: :info)
+
```

Comment:
Just for my understanding, calling [`claims_v1_logging`](https://github.com/department-of-veterans-affairs/vets-api/blob/8eabc0ecb4eb01f7aa6db6d37b1d9d6cb14655f4/modules/claims_api/app/controllers/claims_api/v1/application_controller.rb#L213) here results in an infinite loop because the default `icn` parameter, `target_veteran&.mpi&.icn`, won't exist?
1. How does this result in an infinite loop? Wouldn't it raise a `NameError` or `NoMethodError`?
2. Would it be cleaner to pass `icn: nil` to `claims_v1_logging`, with a comment that warns us not to remove it?

Reply:
@tycol7  When the logger gets called it calls the `target_veteran` method which calls`veteran_from_headers`  and since that errors out in this scenario it then triggers the logger line to run which calls `target_veteran`...

Your solution of adding in `icn: nil` also avoids that.  But for me I just really did not want to call that method for any reason given the scenario.